controller/lib/hal/hal.h

```diff
@@ -75,17 +108,32 @@ extern HalApi Hal;
 
 #ifdef AVR
 
-#include <Arduino.h>
-
 inline uint32_t HalApi::millis() { return ::millis(); }
 inline void HalApi::delay(uint32_t ms) { ::delay(ms); }
 inline int HalApi::analogRead(int pin) { return ::analogRead(pin); }
+inline void HalApi::setDigitalPinMode(int pin, PinMode mode) {
+  ::pinMode(pin, static_cast<uint8_t>(mode));
```

Comment:
nit, I think you could write this as `uint8_t{mode}`.  That form does not allow narrowing casts, so it's slightly safer.

Reply:
Oh nice, thanks, will fix before merging.

Reply:
Huh unfortunately this doesn't seem to work with an enum class. I'm gonna leave as is for now.